packages/admin-date-time/src/core/DatePicker.tsx

```diff
@@ -0,0 +1,148 @@
+    const [customValue, setCustomValue] = React.useState<string | null>(null);
+    const formattedValue = moment(input.value).isValid() ? moment(input.value).format(localeDateFormat) : "";
+    const displayValue = customValue ? customValue : formattedValue;
+
+    const showPicker = () => {
+        onFocus();
+        setShowPopper(true);
+
+        // Wait for the popper to be rendered, so the picker-height can be calculated
+        // correctly https://github.com/airbnb/react-dates/issues/46#issuecomment-255059933
+        setTimeout(() => setShowDayPicker(true), 0);
+    };
+
+    const hidePicker = () => {
+        onBlur();
+        setShowPopper(false);
+        setShowDayPicker(false);
+    };
+
+    const onInputBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+        const newValue = e.currentTarget.value;
+
+        if (newValue) {
+            const momentNewValue = moment(newValue, localeDateFormat);
+
+            if (momentNewValue.isValid()) {
+                onChange(momentNewValue.toDate());
+                setCustomValue(null);
+            } else {
+                onChange(null);
+                setCustomValue(null);
+            }
+        }
+    };
+
+    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const newValue = e.currentTarget.value;
+        if (newValue) {
+            setCustomValue(newValue);
+        } else {
+            onChange(null);
+            setCustomValue(null);
+        }
+    };
+
+    const onDateChange: DayPickerSingleDateControllerShape["onDateChange"] = (date) => {
+        onChange(date ? date.toDate() : null);
+        hidePicker();
+    };
+
+    const rootClasses: string[] = [classes.root];
+    if (disabled) rootClasses.push(classes.disabled);
+    if (fullWidth) rootClasses.push(classes.fullWidth);
+
+    return (
+        <ClickAwayListener onClickAway={hidePicker}>
+            <div ref={rootRef} className={rootClasses.join(" ")}>
+                <InputBase
+                    classes={{ root: classes.inputBase }}
+                    endAdornment={showClearButton ? <ClearInputButton onClick={() => onChange(null)} disabled={!value} /> : undefined}
+                    disabled={disabled}
+                    placeholder={placeholderText}
+                    value={displayValue}
+                    onFocus={showPicker}
+                    onBlur={onInputBlur}
+                    onChange={onInputChange}
+                    inputProps={{
+                        autoComplete: "off",
+                        onKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => {
+                            if (e.key === "Tab") {
+                                // Hide picker, if tab is pressed. This cannot be done in `onBlur` because the input-blur-event is also called,
+                                // when clicking inside the picker and the picker would be hidden, before `onDateChange` could be called.
+                                hidePicker();
```

Comment:
Where did you get this hack from? Is this the only way that needs to be supported?

Reply:
This is needed because Tab would move the focus inside the picker/calendar, instead of the next input in the form.
This way the form can be easily navigated only using the keyboard.